internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -107,19 +107,23 @@ var (
 )
 
 type AgentServiceConfigReconcileContext struct {
-	client.Client
 	Log    logrus.FieldLogger
 	Scheme *runtime.Scheme
 
 	// selector and tolerations the Operator runs in and propagates to its deployments
 	NodeSelector map[string]string
 	Tolerations  []corev1.Toleration
 
+	/* properties. use this field for cross cluster communication */
+	Properties map[string]interface{}
```

Comment:
I'm not really sure what this is, but I don't think maps can be safely used concurrently without locking.
Maybe make this a separate PR with more clear justification?

Reply:
why does the map moved from HASC to the global scope?